pkg/virt-controller/watch/vm_test.go

```diff
@@ -3558,6 +3558,13 @@ var _ = Describe("VirtualMachine", func() {
 
 			It("should reject the request if a VirtualMachineInstancetype conflicts with the VirtualMachineInstance", func() {
 
+				expectedInstancetypeRevisionName := instancetype.GetRevisionName(vm.Name, f.Name, f.UID, f.Generation)
+				expectedInstancetypeRevision, err := instancetype.CreateInstancetypeControllerRevision(vm, expectedInstancetypeRevisionName, f.TypeMeta.APIVersion, &f.Spec)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedRevisionNamePatch, err := instancetype.GenerateRevisionNamePatch(expectedInstancetypeRevision, nil)
+				Expect(err).ToNot(HaveOccurred())
+
```

Comment:
I didn't catch this the first time I looked at this PR.

previously, we'd never store the controller revision unless it could be successfully applied to the vm. Do you think it's important to maintain this behavior? If the webhook validator already performs this check at admission time, then it might not be terribly important to do it again here before storing the revision

Reply:
> I didn't catch this the first time I looked at this PR.
> 
> previously, we'd never store the controller revision unless it could be successfully applied to the vm. Do you think it's important to maintain this behavior? If the webhook validator already performs this check at admission time, then it might not be terribly important to do it again here before storing the revision

Yeah apologies I did mean to expand the comment in the commit message about this but lost context over the long weekend here in the UK. You are correct that we do have an existing check in the validating webhook ensuring things apply cleaning before the VM is accepted [1]. The test here is asserting the behaviour of the controller during application of an instancetype when starting a VM [2].

There's still a tiny chance of a race between the validating webhook checking an instancetype, someone updating the instancetype in a way that could cause a conflict and the controller then capturing a now invalid instancetype in a ControllerRevision when it first sees the VM. As a result I think this test is still valid however unlikely it is to ever actually happen. 

[1] https://github.com/kubevirt/kubevirt/blob/099709e16f4b23507cd3551c115dbe56a5af3c30/pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go#L102-L105
[2] https://github.com/kubevirt/kubevirt/blob/ca6a597a9c2e2e6060a721ecbecbb5660a6b49a5/pkg/virt-controller/watch/vm.go#L978-L985

Reply:
@davidvossel does my comment clarify things?

Reply:
> @davidvossel does my comment clarify things?

yeah, i get the logic. My point was that we could potentially store a revision that can't be applied to the VM since we never verify the revision we're about to store can successfully be applied to the VM. it's an unlikely scenario, but possible now that the revision is kind of "blindly" stored before the controller validates that it applies cleanly.